CMakeLists.txt

```diff
@@ -1825,23 +1825,48 @@ elseif(APPLE)
   endif()
 
   if(SDL_JOYSTICK)
+    file(GLOB MFI_JOYSTICK_SOURCES ${SDL2_SOURCE_DIR}/src/joystick/iphoneos/*.m)
     if(IOS OR TVOS)
-      file(GLOB JOYSTICK_SOURCES ${SDL2_SOURCE_DIR}/src/joystick/iphoneos/*.m ${SDL2_SOURCE_DIR}/src/joystick/steam/*.c)
-      set(SDL_JOYSTICK_MFI 1)
+      file(GLOB JOYSTICK_SOURCES ${SDL2_SOURCE_DIR}/src/joystick/steam/*.c)
+      set(HAVE_FRAMEWORK_GAMECONTROLLER 1)
+      set(HAVE_FRAMEWORK_COREHAPTICS 1)
       if(IOS)
         set(SDL_FRAMEWORK_COREMOTION 1)
       endif()
-      set(SDL_FRAMEWORK_GAMECONTROLLER 1)
-      set(SDL_FRAMEWORK_COREHAPTICS 1)
       set(HAVE_SDL_SENSORS 1)
     else()
-      file(GLOB JOYSTICK_SOURCES ${SDL2_SOURCE_DIR}/src/joystick/darwin/*.c ${SDL2_SOURCE_DIR}/src/joystick/iphoneos/*.m)
-      # FIXME: add checks for SDL_JOYSTICK_MFI???
+      file(GLOB JOYSTICK_SOURCES ${SDL2_SOURCE_DIR}/src/joystick/darwin/*.c)
+      set_source_files_properties(${MFI_JOYSTICK_SOURCES} PROPERTIES COMPILE_FLAGS -fobjc-weak)
+      check_objc_source_compiles("
+          #include <AvailabilityMacros.h>
+          #include <TargetConditionals.h>
+          #import <Foundation/Foundation.h>
+          #import <GameController/GameController.h>
+          #if MAC_OS_X_VERSION_MIN_REQUIRED < 1080
+          #error GameController framework doesn't work on this configuration
+          #endif
+          #if TARGET_CPU_X86
+          #error GameController framework doesn't work on this configuration
+          #endif
+          int main() {}" HAVE_FRAMEWORK_GAMECONTROLLER)
+      check_objc_source_compiles("
+          #include <AvailabilityMacros.h>
+          #include <TargetConditionals.h>
+          #import <Foundation/Foundation.h>
+          #import <CoreHaptics/CoreHaptics.h>
+          int main() {}" HAVE_FRAMEWORK_COREHAPTICS)
       set(SDL_JOYSTICK_IOKIT 1)
       set(SDL_FRAMEWORK_IOKIT 1)
       set(SDL_FRAMEWORK_FF 1)
     endif()
-    set(SOURCE_FILES ${SOURCE_FILES} ${JOYSTICK_SOURCES})
+    if(HAVE_FRAMEWORK_GAMECONTROLLER)
+      set(SDL_JOYSTICK_MFI 1)
+      set(SDL_FRAMEWORK_GAMECONTROLLER 1)
+      if(HAVE_FRAMEWORK_COREHAPTICS)
```

Comment:
Looking at the CI build logs, `EXTRA_LDFLAGS` has:
`-Wl,-weak_framework,QuartzCore;-Wl,-weak_framework,CoreHaptics`

.. in addition, `EXTRA_LIBS` has:
`[....]/MacOSX11.1.sdk/System/Library/Frameworks/GameController.framework`

We don't need the second one, do we not?

Reply:
OK, my bad. We do need the second one, but we should change that `if(SDL_FRAMEWORK_GAMECONTROLLER)` line (at line 2031 in original file) to look like the `if(SDL_FRAMEWORK_COREHAPTICS)` case, I _think_.

Reply:
Agreed, made that change in the latest push.